includes/class-wc-payments.php

```diff
@@ -492,8 +492,9 @@ public static function add_plugin_links( $links ) {
 	 * @return array The list of payment gateways that will be available, including WooCommerce Payments' Gateway class.
 	 */
 	public static function register_gateway( $gateways ) {
-		$gateways[]         = self::$legacy_card_gateway;
-		$reusable_methods[] = self::$legacy_card_gateway;
+		$gateways[]       = self::$legacy_card_gateway;
+		$all_upe_gateways = [];
+		$reusable_methods = [];
 
 		if ( WC_Payments_Features::is_upe_enabled() ) {
 			foreach ( self::$card_gateway->get_payment_method_ids_enabled_at_checkout() as $payment_method_id ) {
```

Comment:
```suggestion
		$gateways[]       = self::$legacy_card_gateway;

		if ( WC_Payments_Features::is_upe_enabled() ) {
	  		$all_upe_gateways = [];
    		$reusable_methods = [];
			foreach ( self::$card_gateway->get_payment_method_ids_enabled_at_checkout() as $payment_method_id ) {
```
Not that important, but maybe these should be defined inside the if UPE block.